pymc3/backends/ndarray.py

```diff
@@ -100,8 +101,10 @@ def save(self, ndarray):
         if not isinstance(ndarray, NDArray):
             raise TypeError('Can only save NDArray')
 
-        if not os.path.exists(self.directory):
-            os.mkdir(self.directory)
+        if os.path.isdir(self.directory):
```

Comment:
Should we ask or alert before removing directories?

Reply:
This scares me too, but my intention here is to treat `self.directory` as a single filesystem object that pymc3 owns.  In this context, it is not unusual to overwrite without asking/alerting.  I think this is fine, but I could see something ugly happening if you call `pm.save_trace(trace, './')`, intending to save a file to the current directory.

My best "safe" alternative that is not overly onerous is to write `.pymc.manifest` to the top level of the directory, with a list of all the files that `pymc` is allowed to delete in the future.  Then this deletion will be just running over all the files in the existing manifest and deleting those.  In that case, I would throw an exception then if any path already exists after deleting files in the manifest.

Reply:
Even more explicitly, by default, there would be a file `./.pymc3.trace/.pymc.manifest`, whose contents are something like

```
0/samples.npz
0/metadata.json
1/samples.npz
1/metadata.json
...
```

Reply:
that's reasonable, though I always appreciate programs where `save` "just works".  what if i followed `jupyter`'s lead and did `.pymc3_1.trace`, `.pymc3_2.trace`, etc. by default?

Reply:
I would imagine that this saving is done multiple times for a single model, so overwriting is probably a useful feature.

Reply:
sorry, i agree with that!  I am suggesting that the signature be 
```
def save_trace(trace, filename=None, overwrite=False):

if filename is None:
    # find a unique filename to write to 
else:
    if filename exists:
        if overwrite is False:
            raise Exception
        else:
            # clear out directory
```

Reply:
ok, that sounds great!